src/gui/accountsettings.cpp

```diff
@@ -243,82 +243,116 @@ void AccountSettings::slotCustomContextMenuRequested(const QPoint &pos)
     const auto removeFolderAction = [this](QMenu *menu) {
         return menu->addAction(tr("Remove folder sync connection"), this, &AccountSettings::slotRemoveCurrentFolder);
     };
+
     QTreeView *tv = ui->_folderList;
     QModelIndex index = tv->indexAt(pos);
     if (!index.isValid()) {
         return;
-    } else if (!_model->data(index, FolderStatusDelegate::IsReady).toBool()) {
+    }
+
+    auto classification = _model->classify(index);
+    if (classification != FolderStatusModel::RootFolder && classification != FolderStatusModel::SubFolder) {
+        return;
+    }
+
+
+    // Only allow removal if the item isn't in "ready" state.
+    if (classification == FolderStatusModel::RootFolder && !_model->data(index, FolderStatusDelegate::IsReady).toBool()) {
         QMenu *menu = new QMenu(tv);
```

Comment:
So only the root node will get the remove action?
(Do we paint the children if the root is != ready?)

Reply:
> So only the root node will get the remove action?

Yes. My thinking was that it's a destructive action, and shouldn't be confused with "delete this folder". But I can change that if you think having it everywhere is more consistent.

> (Do we paint the children if the root is != ready?)

No idea, because I'm not 100% sure what this state means. When the account is singled out it is still ready, so I don't know how to get into that state.

Reply:
A read only folder should be != ready

Reply:
No, the child items are not shown, only the sync root